plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/components/size-control.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import {
+	RangeControl,
+	BaseControl,
+	useBaseControlProps,
+	__experimentalUnitControl as UnitControl,
+	__experimentalGrid as Grid,
+} from '@wordpress/components';
+
+type SizeControlProps = {
+	onChange: ( numbericSize: number, unit: string ) => void;
+	value: string;
+	units: { value: string; label: string; default: number }[];
+	label?: string;
+};
+
+export default function SizeControl( {
+	onChange,
+	value,
+	units,
+	label,
+	...baseProps
+}: SizeControlProps ) {
+	const { baseControlProps } = useBaseControlProps( baseProps );
+
+	const [ numericSize, setNumericSize ] = useState< number >(
+		parseInt( value || '16px', 10 )
+	);
+
+	const [ unit, setUnit ] = useState< string >(
+		value ? value.replace( /[^a-zA-Z]/g, '' ) : 'px'
+	);
+
+	useEffect( () => {
+		onChange( numericSize, unit );
+	}, [ numericSize, unit, onChange ] );
```

Comment:
I believe it would be better to avoid using ‘useEffect’ to trigger the ‘onChange’ function here. Doing so can cause performance issues because every time the component re-renders, the ‘onChange’ function is created again. When React performs a shallow comparison of the array that is passed, it will not be the same, and this will re-trigger multiple times. Sometimes, this can even cause infinite loops. 

```suggestion
```

You can call the `onChange` function on each place that you are changing the "numericSize" and/or the "unit" (I'll leave some examples in the code below).